src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -558,6 +565,7 @@ private void readOffset() throws ConnectException {
     );
     if (fileStatusWithMaxOffset != null) {
       offset = FileUtils.extractOffset(fileStatusWithMaxOffset.getPath().getName()) + 1;
+      committedOffset = offset;
```

Comment:
If we read the offset from HDFS files, we update the committed offset here.

Reply:
nit: the committedOffset is actually `offset - 1`. `offset` here is the value we will next read from.

Reply:
(pressed resolved by accident, apologies). 
Setting 
`committedOffset = FileUtils.extractOffset(fileStatusWithMaxOffset.getPath().getName());`
and then
`offset = committedOffset + 1`
probably makes more sense (fyi `preCommit` won't accidentally commit offsets that have not been consumed, we might delay offset commit with a wrong setting).

Reply:
I just made this same change re setting `committedOffset` and `offset`. Great minds... :-D

What do you mean by "we might delay offset commit with a wrong setting"?

Reply:
I meant that there's no risk to commit offsets ahead of the current position with `preCommit`. 
See: https://github.com/apache/kafka/blob/trunk/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java#L410-L421

It's just that the recommendation will be ignored and therefore the lag won't be represented as soon as it would be possible.

Reply:
Gotcha. The WorkerSinkTask will prevent committing offsets for records that haven't yet been consumed, but IMO it'd be better to not rely upon that feature since it may result in a larger lag (as measured by the committed offset), especially if the connector always returns a yet-to-be-consumed offset.

I think we're all on the same page, and the updated code uses the correct offset anyway.